cli/src/action/circuit/mod.rs

```diff
@@ -235,6 +263,34 @@ fn parse_application_metadata_json(metadata: &str) -> Result<String, CliError> {
     Ok(format!("\"{}\":{}", key, value))
 }
 
+#[cfg(feature = "circuit-template")]
+fn parse_template_args(args: &[&str]) -> Result<HashMap<String, String>, CliError> {
+    args.iter().try_fold(HashMap::new(), |mut acc, arg| {
+        let mut iter = arg.split('=');
+        let key = iter
+            .next()
+            .ok_or_else(|| {
+                CliError::ActionError(format!(
+                    "Invalid template argument. Expected value in form <key>=<value> found {}",
+                    arg
+                ))
+            })?
+            .to_string();
```

Comment:
Might also want to check that these are non-empty.

Reply:
addressed